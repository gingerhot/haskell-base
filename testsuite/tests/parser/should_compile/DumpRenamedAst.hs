{-# LANGUAGE DataKinds, PolyKinds, TypeOperators, TypeFamilies #-}

module DumpRenamedAst where

data Peano = Zero | Succ Peano

type family Length (as :: [k]) :: Peano where
  Length (a : as) = Succ (Length as)
  Length '[]      = Zero

main = putStrLn "hello"
