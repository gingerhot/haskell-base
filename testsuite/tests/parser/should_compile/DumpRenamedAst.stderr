
==================== Renamer ====================

(HsGroup 
 (ValBindsOut 
  [
   ((,) 
    (NonRecursive) {Bag(Located (HsBind Name)): 
    [
     ({ DumpRenamedAst.hs:11:1-23 }
      (FunBind 
       ({ DumpRenamedAst.hs:11:1-4 }{Name: main:DumpRenamedAst.main{v}}) 
       (MG 
        ({ DumpRenamedAst.hs:11:1-23 }
         [
          ({ DumpRenamedAst.hs:11:1-23 }
           (Match 
            (FunRhs 
             ({ DumpRenamedAst.hs:11:1-4 }{Name: main:DumpRenamedAst.main{v}}) 
             (Prefix) 
             (NoSrcStrict)) 
            [] 
            (Nothing) 
            (GRHSs 
             [
              ({ DumpRenamedAst.hs:11:6-23 }
               (GRHS 
                [] 
                ({ DumpRenamedAst.hs:11:8-23 }
                 (HsApp 
                  ({ DumpRenamedAst.hs:11:8-15 }
                   (HsVar 
                    ({ DumpRenamedAst.hs:11:8-15 }{Name: base:System.IO.putStrLn{v}}))) 
                  ({ DumpRenamedAst.hs:11:17-23 }
                   (HsLit 
                    (HsString 
                     (SourceText "\"hello\"") {FastString: "hello"})))))))] 
             ({ <no location info> }
              (EmptyLocalBinds)))))]) 
        [] 
        (PlaceHolder) 
        (FromSource)) 
       (WpHole) {NameSet: 
       []} 
       []))]})] 
  []) 
 [] 
 [
  (TyClGroup 
   [
    ({ DumpRenamedAst.hs:5:1-30 }
     (DataDecl 
      ({ DumpRenamedAst.hs:5:6-10 }{Name: main:DumpRenamedAst.Peano{tc}}) 
      (HsQTvs 
       [] 
       [] {NameSet: 
       []}) 
      (Prefix) 
      (HsDataDefn 
       (DataType) 
       ({ <no location info> }
        []) 
       (Nothing) 
       (Nothing) 
       [
        ({ DumpRenamedAst.hs:5:14-17 }
         (ConDeclH98 
          ({ DumpRenamedAst.hs:5:14-17 }{Name: main:DumpRenamedAst.Zero{d}}) 
          (Nothing) 
          (Just 
           ({ <no location info> }
            [])) 
          (PrefixCon 
           []) 
          (Nothing))),
        ({ DumpRenamedAst.hs:5:21-30 }
         (ConDeclH98 
          ({ DumpRenamedAst.hs:5:21-24 }{Name: main:DumpRenamedAst.Succ{d}}) 
          (Nothing) 
          (Just 
           ({ <no location info> }
            [])) 
          (PrefixCon 
           [
            ({ DumpRenamedAst.hs:5:26-30 }
             (HsTyVar 
              (NotPromoted) 
              ({ DumpRenamedAst.hs:5:26-30 }{Name: main:DumpRenamedAst.Peano{tc}})))]) 
          (Nothing)))] 
       ({ <no location info> }
        [])) 
      (True) {NameSet: 
      [{Name: main:DumpRenamedAst.Peano{tc}}]}))] 
   [] 
   []),
  (TyClGroup 
   [
    ({ DumpRenamedAst.hs:7:1-39 }
     (FamDecl 
      (FamilyDecl 
       (ClosedTypeFamily 
        (Just 
         [
          ({ DumpRenamedAst.hs:8:3-36 }
           (TyFamEqn 
            ({ DumpRenamedAst.hs:8:3-8 }{Name: main:DumpRenamedAst.Length{tc}}) 
            (HsIB 
             [{Name: a{tv}},{Name: as{tv}}] 
             [
              ({ DumpRenamedAst.hs:8:10-17 }
               (HsParTy 
                ({ DumpRenamedAst.hs:8:11-16 }
                 (HsOpTy 
                  ({ DumpRenamedAst.hs:8:11 }
                   (HsTyVar 
                    (NotPromoted) 
                    ({ DumpRenamedAst.hs:8:11 }{Name: a{tv}}))) 
                  ({ DumpRenamedAst.hs:8:13 }{Name: ghc-prim:GHC.Types.:{(w) d}}) 
                  ({ DumpRenamedAst.hs:8:15-16 }
                   (HsTyVar 
                    (NotPromoted) 
                    ({ DumpRenamedAst.hs:8:15-16 }{Name: as{tv}})))))))] 
             (True)) 
            (Prefix) 
            ({ DumpRenamedAst.hs:8:21-36 }
             (HsAppTy 
              ({ DumpRenamedAst.hs:8:21-24 }
               (HsTyVar 
                (NotPromoted) 
                ({ DumpRenamedAst.hs:8:21-24 }{Name: main:DumpRenamedAst.Succ{d}}))) 
              ({ DumpRenamedAst.hs:8:26-36 }
               (HsParTy 
                ({ DumpRenamedAst.hs:8:27-35 }
                 (HsAppTy 
                  ({ DumpRenamedAst.hs:8:27-32 }
                   (HsTyVar 
                    (NotPromoted) 
                    ({ DumpRenamedAst.hs:8:27-32 }{Name: main:DumpRenamedAst.Length{tc}}))) 
                  ({ DumpRenamedAst.hs:8:34-35 }
                   (HsTyVar 
                    (NotPromoted) 
                    ({ DumpRenamedAst.hs:8:34-35 }{Name: as{tv}}))))))))))),
          ({ DumpRenamedAst.hs:9:3-24 }
           (TyFamEqn 
            ({ DumpRenamedAst.hs:9:3-8 }{Name: main:DumpRenamedAst.Length{tc}}) 
            (HsIB 
             [] 
             [
              ({ DumpRenamedAst.hs:9:10-12 }
               (HsExplicitListTy 
                (Promoted) 
                (PlaceHolder) 
                []))] 
             (True)) 
            (Prefix) 
            ({ DumpRenamedAst.hs:9:21-24 }
             (HsTyVar 
              (NotPromoted) 
              ({ DumpRenamedAst.hs:9:21-24 }{Name: main:DumpRenamedAst.Zero{d}})))))])) 
       ({ DumpRenamedAst.hs:7:13-18 }{Name: main:DumpRenamedAst.Length{tc}}) 
       (HsQTvs 
        [{Name: k{tv}}] 
        [
         ({ DumpRenamedAst.hs:7:20-30 }
          (KindedTyVar 
           ({ DumpRenamedAst.hs:7:21-22 }{Name: as{tv}}) 
           ({ DumpRenamedAst.hs:7:27-29 }
            (HsListTy 
             ({ DumpRenamedAst.hs:7:28 }
              (HsTyVar 
               (NotPromoted) 
               ({ DumpRenamedAst.hs:7:28 }{Name: k{tv}})))))))] {NameSet: 
        []}) 
       (Prefix) 
       ({ DumpRenamedAst.hs:7:32-39 }
        (KindSig 
         ({ DumpRenamedAst.hs:7:35-39 }
          (HsTyVar 
           (NotPromoted) 
           ({ DumpRenamedAst.hs:7:35-39 }{Name: main:DumpRenamedAst.Peano{tc}}))))) 
       (Nothing))))] 
   [] 
   [])] 
 [] 
 [] 
 [] 
 [] 
 [] 
 [] 
 [] 
 [] 
 [])

